core/src/main/java/org/testcontainers/utility/ResourceReaper.java

```diff
@@ -34,6 +60,94 @@ private ResourceReaper() {
         Runtime.getRuntime().addShutdownHook(new Thread(this::performCleanup));
     }
 
+    @SneakyThrows(InterruptedException.class)
+    public static String start(String hostIpAddress, DockerClient client) {
+        String ryukImage = TestcontainersConfiguration.getInstance().getRyukImage();
+        client.pullImageCmd(ryukImage).exec(new PullImageResultCallback()).awaitSuccess();
+
+        MountableFile mountableFile = MountableFile.forClasspathResource(ResourceReaper.class.getName().replace(".", "/") + ".class");
+
+        String ryukContainerId = client.createContainerCmd(ryukImage)
+                .withHostConfig(new HostConfig() {
+                    @JsonProperty("AutoRemove")
```

Comment:
`docker-java` is behind, but this flag will remove the container if it stops (and it will stop when the cleanup happen)

Reply:
Aha, I was wondering what would remove this container 😄